libs/net/dal/TNOContext.cs

```diff
@@ -143,19 +145,40 @@ public override int SaveChanges()
     public int CommitTransaction()
     {
         var result = 0;
-        using (var transaction = this.Database.BeginTransaction())
+        var retry = 0;
+        _saveChangesLock.Wait();
+        try
         {
-            try
+            using var transaction = Database.BeginTransaction();
+            while (result == 0 && retry < _maxRetry)
             {
-                result = this.SaveChanges();
-                transaction.Commit();
-            }
-            catch (DbUpdateException)
-            {
-                transaction.Rollback();
-                throw;
+                try
+                {
+                    result = SaveChanges();
+                    transaction.Commit();
+                }
+                catch (DbUpdateConcurrencyException ex)
+                {
+                    retry++;
+                    _logger?.LogError(ex, $"{nameof(TNOContext)} caught {nameof(DbUpdateConcurrencyException)} {retry}");
```

Comment:
You can't do this it defeats the purpose of a concurrency error.  A concurrency error means someone is attempting to overwrite a newer record with an older one.  The issue we need to solve is first find out what is attempting to save an old version of a row.  Then find out how to ensure it has the latest version before saving.

Reply:
Updated for gathering more info about the potential causes, thanks